basic-services/README.md

```diff
@@ -0,0 +1,176 @@
+Only the smallest of commits should be one line messages.
+
+* Consider adding a light weight CONTRIBUTING.md or referencing Microsoft's for guidance for outsiders and others in JCI.
+
+* Track the Issues on this repo and provide guidance (where possible) on roadmap.
+
+### Phase 1
+
+We will begin on what I'm calling the TraditionalClient. (name can be changed).
+I'm calling it the TraditionalClient because of its heavy influence on the old MSSDA.
+The traditional client is probably ideal for the typical MSSDA customer because it 
+doesn't utilize many advanced C# features. But it is non-ideal for advanced development.
+
+1. It is synchronous. A more advanced SDK would expose all operations as asynch operations.
+2. It doesn't use generics. Instead it returns simple return structures (like ReadPropertyResult) that
+   model attribute values as strings, numbers and array. The client just needs to know based on what
+   attribute they are reading which of the values is most appropriate for thier usage. This is how 
+   MSSDA worked as well. A more advanced SDK would use `JToken` values or `dynamic` or generics.
+3. It doesn't use Nullable<T> which makes it a little awkward in some cases. Like what is the numeric value
+   of the string "AV1"? Using the MSSDA guidelines the numeric value of a string will be 0. (A more sophisticated
+   API would mark the numeric value as nullable for when it doesn't make sense)
+   
+The following requirements will be met in phase 1:
+
+1. Implement Login
+1. Implement GetObjectIdentifier
+1. Implement ReadProperty
+2. Implement ReadPropertyMultiple
+3. Expose the library as COM dll for use in Excel apps
+4. Ensure signatures of the methods are easy to consume via COM
+5. Must provide client side resource files for translating enum values
+   coming back from the server. (Unlike MSSDA, enums are not returned
+   from the server as localized strings. They are returned as "symbolic
+   strings" which are more useful at the API level. The client side
+   resource files will provide the ability for clients to show localized
+   strings.
+6. Nuget package support (for deployment to metasys-server org)
+7. Script for registering dll for COM (would only be used by Excel apps, wouldn't be needed for .Net apps)
+
+Suppoted Data Types
+
+boolean, numbers, strings, enums, arrays of one of these.
+
+Not supported: date, time, listOf, structs, object identifiers, bits, etc.
+
+### Phase 2
+
+Allow for reverse lookup of resource file. Given a localized string,
+find the enum value associated (for usage in SendCommand and WriteProperty)
+
+2. Implement WriteProperty
+4. Implement WritePropertyMultiple
+5. Implement GetCommands
+5. Implement SendCommand
+
+
+### Phase 3
+
+1. Implement GetObjectList() this will differ from MSSDA. MSSDA returned just the immediate children of given object.
+   10.1 Metasys Server has the /objects endpoint which allows you to specify a parent object and ask for children `/objects/{id}/objects`
```

Comment:
Technically, spaces and equipment are objects, so they'll be returned (they may be hidden, not sure what happens by default).  Does this complicate these use cases any?   Do we want them returned?

Either way, not a new problem, but it does bring in consistency questions with the UI and more authorization complexity.

Reply:
I think It's a don't care. If they want to read one of those and the API supports it, so will the SDK.